### ISSUE TITLE
Support for shiny input in `colDef(cell = function(i) {...})`

### ISSUE THREAD

**Issue by eheinzen** [glin/reactable#390]
In #89 you mentioned that Shiny inputs are not yet supported in the cell definition, but are supported in the details.  I've got a use case where I want the rows of my table to have a numericInput associated with them, but I don't want to put it in the details section.  Any chance `reactable` could support that?

I'm looking into a workaround with Javascript for now, but I'm definitely no JS expert...

**Comment by glin** [comment]
Yeah, this is still on the to-do list somewhere. The last update was at https://github.com/glin/reactable/issues/345#issuecomment-1794047821, where I felt like it wouldn't be a common enough use case. Another issue was that it'd probably incur a performance cost to look for and re-bind Shiny inputs on every table change, even if there weren't any Shiny inputs present. I was thinking about making this more of an opt-in feature where you'd need to mark which cells could contain Shiny input/outputs.

For now, a possibility could be to use [`Reactable.onStateChange()`](https://glin.github.io/reactable/articles/javascript-api.html#reactable-onstatechange) to re-bind Shiny inputs on every table change manually.

This would be something like:
```js
Reactable.onStateChange('table-id', state => {
  Shiny.unbindAll()
  Shiny.bindAll()
})
```

I haven't tested this, so I'm not 100% sure it'll work.
